src/main/java/org/folio/circulation/storage/ItemByBarcodeInStorageFinder.java

```diff
@@ -1,38 +1,30 @@
 package org.folio.circulation.storage;
 
+import static org.folio.circulation.domain.validation.CommonFailures.noItemFoundForBarcodeFailure;
 import static org.folio.circulation.support.Result.of;
 
 import java.util.concurrent.CompletableFuture;
-import java.util.function.Supplier;
 
 import org.folio.circulation.domain.Item;
-import org.folio.circulation.support.HttpFailure;
-import org.folio.circulation.support.Result;
 import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.Result;
 
 public class ItemByBarcodeInStorageFinder {
   private final ItemRepository itemRepository;
-  private final Supplier<HttpFailure> itemNotFoundFailureSupplier;
```

Comment:
Why has the failure supplier been removed?

Reply:
I have modified `CheckInAdapter` class to construct all dependencies internally and in order to not have dependency on particular item, I have replaced the `itemNotFoundFailureSupplier` with direct call of `noItemFoundForBarcodeFailure(...)` function.

Reply:
Ok, this may well be over-design from when check-in was intended to be allowed via barcode or `id`.

Reply:
Were all of the supplier using the `noItemFoundForBarcodeFailure` indirectly?

Reply:
Yes, there are 2 places where it is used:
![image](https://user-images.githubusercontent.com/9094650/74738600-ebd1fe00-525f-11ea-9cdb-0fe91e56c74b.png)

* Check-in
![image](https://user-images.githubusercontent.com/9094650/74738625-fee4ce00-525f-11ea-809a-ea04da2b2b26.png)
* find-loan
![image](https://user-images.githubusercontent.com/9094650/74738653-0f954400-5260-11ea-9955-824ebc788d66.png)